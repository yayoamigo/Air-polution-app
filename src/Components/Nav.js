import React from 'react';
import { Link } from 'react-router-dom';
import {BsFillMicFill, BsGearFill} from 'react-icons/bs';


export const Nav = () => {

  return (
    <header>
     <nav className='navBar'>
      <button type='button'>
       <Link className="link" to="/"> 
        <i class="fa-solid fa-less-than"></i>
       </Link>  
      </button>
      <h1> Air polution app</h1>
      <div className='sideBar'>
      <BsFillMicFill/>
      <BsGearFill/>
      </div>
     </nav>
    </header>
  )
}
