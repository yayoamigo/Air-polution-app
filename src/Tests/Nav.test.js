import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Nav from '../Components/Nav';
import store from '../Redux/configureStore';

it('render', () => {
  const tree = renderer
    .create(
      <Router>
        <Provider store={store}>
          <Nav />
        </Provider>
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders content', () => {
  const component = render(
    <Router>
      <Provider store={store}>
        <Nav />
      </Provider>
    </Router>,
  );
  expect(component.container).toHaveTextContent('Air polution app');
});
