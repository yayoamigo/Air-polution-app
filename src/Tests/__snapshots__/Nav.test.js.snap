// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<header>
  <nav
    className="navBar"
  >
    <button
      type="button"
    >
      <a
        className="link"
        href="/"
        onClick={[Function]}
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="19"
            x2="5"
            y1="12"
            y2="12"
          />
          <polyline
            points="12 19 5 12 12 5"
          />
        </svg>
      </a>
    </button>
    <h1>
       Air polution app
    </h1>
    <div
      className="sideBar"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5 3a3 3 0 0 1 6 0v5a3 3 0 0 1-6 0V3z"
        />
        <path
          d="M3.5 6.5A.5.5 0 0 1 4 7v1a4 4 0 0 0 8 0V7a.5.5 0 0 1 1 0v1a5 5 0 0 1-4.5 4.975V15h3a.5.5 0 0 1 0 1h-7a.5.5 0 0 1 0-1h3v-2.025A5 5 0 0 1 3 8V7a.5.5 0 0 1 .5-.5z"
        />
      </svg>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.405 1.05c-.413-1.4-2.397-1.4-2.81 0l-.1.34a1.464 1.464 0 0 1-2.105.872l-.31-.17c-1.283-.698-2.686.705-1.987 1.987l.169.311c.446.82.023 1.841-.872 2.105l-.34.1c-1.4.413-1.4 2.397 0 2.81l.34.1a1.464 1.464 0 0 1 .872 2.105l-.17.31c-.698 1.283.705 2.686 1.987 1.987l.311-.169a1.464 1.464 0 0 1 2.105.872l.1.34c.413 1.4 2.397 1.4 2.81 0l.1-.34a1.464 1.464 0 0 1 2.105-.872l.31.17c1.283.698 2.686-.705 1.987-1.987l-.169-.311a1.464 1.464 0 0 1 .872-2.105l.34-.1c1.4-.413 1.4-2.397 0-2.81l-.34-.1a1.464 1.464 0 0 1-.872-2.105l.17-.31c.698-1.283-.705-2.686-1.987-1.987l-.311.169a1.464 1.464 0 0 1-2.105-.872l-.1-.34zM8 10.93a2.929 2.929 0 1 1 0-5.86 2.929 2.929 0 0 1 0 5.858z"
        />
      </svg>
    </div>
  </nav>
</header>
`;
